infra/terraform/modules/data_access/memberships/memberships.py

```diff
@@ -114,7 +115,19 @@ def policy_arn(team_slug, policy_type):
 
 
 def bucket_name(team_slug):
-    return "{env}-{team_slug}".format(
-        env=os.environ["STAGE"],
-        team_slug=team_slug.lower()
-    )
+    '''
+    Generate the S3 bucket name by prefixing the environment name and
+    replacing invalid characters with an hyphen ('-').
+
+    NOTE: This is a very simple implementation which doesn't cover all the
+          S3 limitations (e.g. max length or labels limitations, etc...)
+
+    See: http://docs.aws.amazon.com/en_gb/AmazonS3/latest/dev/BucketRestrictions.html
+    '''
+
+    INVALID_BUCKET_CHARS = r"[^a-z0-9.-]"
+
+    name = team_slug.lower()
+    name = re.sub(INVALID_BUCKET_CHARS, "-", name)
```

Comment:
You are right. As said in the function comment, this is a simple implementation which doesn't cover all S3 limitations but should cover all the likely team names without having to store the mapping between GitHub team names and the generated S3 bucket name.

How would the code be automatically be able to transform a team called `1.2.3.4` or `...` into a valid S3 bucket?